diff -Naur -X excludes /Users/matt/Downloads/irssi-0.8.11/src/core/core.h irssi-0.8.11/src/core/core.h
--- /Users/matt/Downloads/irssi-0.8.11/src/core/core.h	2007-04-25 19:15:07.000000000 +0100
+++ irssi-0.8.11/src/core/core.h	2007-11-04 19:01:46.000000000 +0000
@@ -10,6 +10,7 @@
 #define IRSSI_GUI_GNOME	3
 #define IRSSI_GUI_QT   	4
 #define IRSSI_GUI_KDE  	5
+#define IRSSI_GUI_AQUA 	6
 
 extern int irssi_gui;
 extern int irssi_init_finished; /* TRUE after "irssi init finished" signal is sent */
diff -Naur -X excludes /Users/matt/Downloads/irssi-0.8.11/src/core/nicklist.c irssi-0.8.11/src/core/nicklist.c
--- /Users/matt/Downloads/irssi-0.8.11/src/core/nicklist.c	2007-04-25 19:15:07.000000000 +0100
+++ irssi-0.8.11/src/core/nicklist.c	2007-11-04 19:01:46.000000000 +0000
@@ -377,7 +377,7 @@
 	 * returns :-)
 	 * -- yath */
 
-	if (p1->other) {
+	if (p1->other || p1->serverop) {
 		const char *other = (nick_prefix == NULL) ? NULL : strchr(nick_prefix, p1->other);
 		status1 = (other == NULL) ? 5 : 1000 - (other - nick_prefix);
 	} else if (p1->op)
@@ -389,7 +389,7 @@
 	else
 		status1 = 1;
 
-	if (p2->other) {
+	if (p2->other || p2->serverop) {
 		const char *other = (nick_prefix == NULL) ? NULL : strchr(nick_prefix, p2->other);
 		status2 = (other == NULL) ? 5 : 1000 - (other - nick_prefix);
 	} else if (p2->op)
diff -Naur -X excludes /Users/matt/Downloads/irssi-0.8.11/src/fe-common/core/fe-channels.c irssi-0.8.11/src/fe-common/core/fe-channels.c
--- /Users/matt/Downloads/irssi-0.8.11/src/fe-common/core/fe-channels.c	2007-04-25 19:15:07.000000000 +0100
+++ irssi-0.8.11/src/fe-common/core/fe-channels.c	2007-11-04 19:02:09.000000000 +0000
@@ -105,9 +105,11 @@
 
 static void sig_channel_joined(CHANNEL_REC *channel)
 {
+	#if 0
 	if (settings_get_bool("show_names_on_join") &&
 	    !channel->session_rejoin)
 		fe_channels_nicklist(channel, CHANNEL_NICKLIST_FLAG_ALL);
+	#endif
 }
 
 static void cmd_wjoin_pre(const char *data, SERVER_REC *server)
diff -Naur -X excludes /Users/matt/Downloads/irssi-0.8.11/src/fe-common/core/fe-core-commands.c irssi-0.8.11/src/fe-common/core/fe-core-commands.c
--- /Users/matt/Downloads/irssi-0.8.11/src/fe-common/core/fe-core-commands.c	2007-04-25 19:15:07.000000000 +0100
+++ irssi-0.8.11/src/fe-common/core/fe-core-commands.c	2007-11-04 22:24:45.000000000 +0000
@@ -106,8 +106,7 @@
 	if (*data == '\0') {
                 g_snprintf(time, sizeof(time), "%04d", IRSSI_VERSION_TIME);
 		printtext(NULL, NULL, MSGLEVEL_CLIENTNOTICE,
-			  "Client: "PACKAGE" " IRSSI_VERSION" (%d %s)",
-			  IRSSI_VERSION_DATE, time);
+			  "Client: MacIrssi 0.8.2");
 	}
 }
 
diff -Naur -X excludes /Users/matt/Downloads/irssi-0.8.11/src/fe-common/core/fe-help.c irssi-0.8.11/src/fe-common/core/fe-help.c
--- /Users/matt/Downloads/irssi-0.8.11/src/fe-common/core/fe-help.c	2007-04-25 19:15:07.000000000 +0100
+++ irssi-0.8.11/src/fe-common/core/fe-help.c	2007-11-04 19:02:09.000000000 +0000
@@ -265,7 +265,7 @@
 
 void fe_help_init(void)
 {
-        settings_add_str("misc", "help_path", HELPDIR);
+	settings_add_str("misc", "help_path", "Contents/Resources/helpfiles");
 	command_bind("help", NULL, (SIGNAL_FUNC) cmd_help);
 }
 
diff -Naur -X excludes /Users/matt/Downloads/irssi-0.8.11/src/fe-common/core/fe-messages.c irssi-0.8.11/src/fe-common/core/fe-messages.c
--- /Users/matt/Downloads/irssi-0.8.11/src/fe-common/core/fe-messages.c	2007-04-25 19:15:07.000000000 +0100
+++ irssi-0.8.11/src/fe-common/core/fe-messages.c	2007-11-04 19:02:09.000000000 +0000
@@ -565,7 +565,7 @@
                 return;
 
 	firstnick = g_hash_table_lookup(channel->nicks, nick->nick);
-	if (firstnick->next == NULL)
+	if (!firstnick || firstnick->next == NULL)
 		return;
 
 	if (nick == channel->ownnick) {
diff -Naur -X excludes /Users/matt/Downloads/irssi-0.8.11/src/fe-common/core/themes.c irssi-0.8.11/src/fe-common/core/themes.c
--- /Users/matt/Downloads/irssi-0.8.11/src/fe-common/core/themes.c	2007-04-25 19:15:07.000000000 +0100
+++ irssi-0.8.11/src/fe-common/core/themes.c	2007-11-04 19:02:09.000000000 +0000
@@ -38,6 +38,8 @@
 GHashTable *default_formats;
 
 static int init_finished;
+int num_theme_dirs;
+char **theme_dirs;
 static char *init_errors;
 static THEME_REC *internal_theme;
 
@@ -840,10 +842,24 @@
 		g_free(fname);
 		fname = g_strdup_printf(THEMESDIR"/%s.theme", name);
 		if (stat(fname, &statbuf) != 0) {
-			/* theme not found */
 			g_free(fname);
+
+			/* MacIrssi - check additional dirs */
+			int i, theme_found = 0;
+			for(i = 0; i < num_theme_dirs; i++) {
+				fname = g_strdup_printf("%s/%s.theme", theme_dirs[i], name);
+				if (stat(fname, &statbuf) == 0) {
+					theme_found = 1;
+					break;
+				}
+				g_free(fname);
+			}
+			
+			/* theme not found */
+			if (!theme_found) {
 			g_free(name);
 			return theme; /* use the one in memory if possible */
+			}
 		}
 	}
 
diff -Naur -X excludes /Users/matt/Downloads/irssi-0.8.11/src/fe-common/core/themes.h irssi-0.8.11/src/fe-common/core/themes.h
--- /Users/matt/Downloads/irssi-0.8.11/src/fe-common/core/themes.h	2007-04-25 19:15:07.000000000 +0100
+++ irssi-0.8.11/src/fe-common/core/themes.h	2007-11-04 19:02:09.000000000 +0000
@@ -1,6 +1,8 @@
 #ifndef __THEMES_H
 #define __THEMES_H
 
+extern int num_theme_dirs;
+extern char **theme_dirs;
 typedef struct {
 	char *name;
 
diff -Naur -X excludes /Users/matt/Downloads/irssi-0.8.11/src/fe-common/core/window-activity.c irssi-0.8.11/src/fe-common/core/window-activity.c
--- /Users/matt/Downloads/irssi-0.8.11/src/fe-common/core/window-activity.c	2007-04-25 19:15:07.000000000 +0100
+++ irssi-0.8.11/src/fe-common/core/window-activity.c	2007-11-04 19:02:09.000000000 +0000
@@ -34,6 +34,7 @@
 
 static char **hide_targets;
 static int hide_level, msg_level, hilight_level;
+int currentDataLevel;
 
 void window_activity(WINDOW_REC *window, int data_level,
 		     const char *hilight_color)
@@ -79,8 +80,6 @@
         char *tagtarget;
 	int data_level;
 
-	if (dest->window == active_win || (dest->level & hide_level))
-		return;
 
 	if (dest->level & hilight_level) {
 		data_level = DATA_LEVEL_HILIGHT+dest->hilight_priority;
@@ -88,6 +87,11 @@
 		data_level = (dest->level & msg_level) ?
 			DATA_LEVEL_MSG : DATA_LEVEL_TEXT;
 	}
+	
+	currentDataLevel = data_level;
+	if (dest->window == active_win || (dest->level & hide_level))
+		return;
+	
 
 	if ((dest->level & MSGLEVEL_HILIGHT) == 0 && dest->target != NULL) {
 		/* check for both target and tag/target */
diff -Naur -X excludes /Users/matt/Downloads/irssi-0.8.11/src/irc/core/ctcp.c irssi-0.8.11/src/irc/core/ctcp.c
--- /Users/matt/Downloads/irssi-0.8.11/src/irc/core/ctcp.c	2007-04-25 19:15:07.000000000 +0100
+++ irssi-0.8.11/src/irc/core/ctcp.c	2007-11-04 22:25:18.000000000 +0000
@@ -328,8 +328,7 @@
 {
 	ctcp_cmds = NULL;
 
-	settings_add_str("misc", "ctcp_version_reply",
-			 PACKAGE" v$J - running on $sysname $sysarch");
+	settings_add_str("misc", "ctcp_version_reply", "MacIrssi 0.8.2");
 	settings_add_str("misc", "ctcp_userinfo_reply", "$Y");
 	settings_add_int("flood", "max_ctcp_queue", 5);
 
